### PR TITLE
feat: add gqm_gen script

### PR DESCRIPTION
**Pull request by jeffabailey**
Initial version to keep everyone in the loop.

See linked PR for remaining tasks.

This is a fix for https://github.com/InnerSourceCommons/managing-innersource-projects/issues/37

### CODE REVIEW

File: scripts/gqm_gen/gqm.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+> gqmgen@1.0.0 start
+
+```mermaid
```
Comment (rrrutledge) [inline scripts/gqm_gen/gqm.md:4]: Is this the output?  Maybe put in a comment to note that it is autogenerated.

**Review by rrrutledge — COMMENTED** [review]
So excited for this work?  Only other comment is to add a README to the `scripts` directory before merge.

File: scripts/gqm_gen/gqm.md (hunk 1 above)
Comment (jeffabailey) [reply scripts/gqm_gen/gqm.md:4]: Yes, that's the output.

It shouldn't actually be checked in. :)

I'll definitely add a README.md.

I'll get to buttoning this up soon, maybe in today's mob session.

**Comment by rrrutledge** [comment]
I’ll take a look and try it out!

File: scripts/gqm_gen/index.ts [hunk 2]
```diff
@@ -0,0 +1,190 @@
+import * as Commonmark from "commonmark";
+import * as fs from "fs";
+import * as path from "path";
+import { Node, Edge, Graph, Link, LinkType, NodeShape, ArrowType } from "./types";
+import { FileLink } from "./types";
+
+const graph = getGQMFileLinks();
+
+export function getLinkUrl(linkType: LinkType, file: string) {
+  // https://github.com/InnerSourceCommons/managing-inner-source-projects/blob/main/measuring/goals/reduce-duplication.md
```
Comment (rrrutledge) [inline scripts/gqm_gen/index.ts:10]:
```suggestion
```

**Review by rrrutledge — APPROVED** [review]
So excited for this work‼️

**Pull request merged**